src/ReactTags.js

```diff
@@ -274,7 +274,13 @@ class ReactTags extends Component {
     e.preventDefault();
 
     const clipboardData = e.clipboardData || window.clipboardData;
-    const pastedText = clipboardData.getData('text');
+    const clipboardText = clipboardData.getData('text');
+
+    // Checking if the pasted text is greater than maxLength props
+    const { maxLength = clipboardText.length } = this.props;
+
+    const maxTextLength = Math.min(maxLength, clipboardText.length) || clipboardText.length;
```

Comment:
@imtanmoy in the previous comment I mentioned that only of the suggestions is needed, both will be redundant,
so you can change this to `const maxTextLength = Math.min(maxLength, clipboardText.length)`